src/core/NotePlayHandle.cpp

```diff
@@ -555,15 +557,22 @@ void NotePlayHandle::updateFrequency()
 
 
 
-void NotePlayHandle::processTimePos( const TimePos& time )
+void NotePlayHandle::processTimePos(const TimePos& time, float pvalue)
 {
-	if( detuning() && time >= songGlobalParentOffset()+pos() )
+	if (detuning() && time >= songGlobalParentOffset() + pos())
 	{
-		const float v = detuning()->automationClip()->valueAt( time - songGlobalParentOffset() - pos() );
-		if( !typeInfo<float>::isEqual( v, m_baseDetuning->value() ) )
+		if (GuiApplication::instance()->pianoRoll()->isRecording() && m_origin == Origin::OriginMidiInput)
```

Comment:
To remove `GuiApplication` here we can probably add a default boolean to the function for `isRecording`.